spec/factories/course_discussion_topics.rb

```diff
@@ -6,5 +6,11 @@
     after(:build) do |topic|
       topic.actable = build(:forum_topic, topic: topic) unless topic.actable
     end
+
+    trait :with_post do
+      after(:build) do |topic|
```

Comment:
I think a transient attribute with a post count is more generic, and the post count defaults to 0. If a object needs to have posts, then it can set the count to 1.